src/ucs/memory/rcache.c

```diff
@@ -300,7 +300,9 @@ ucs_rcache_region_lru_add(ucs_rcache_t *rcache, ucs_rcache_region_t *region)
     ucs_rcache_region_trace(rcache, region, "lru add");
     ucs_list_add_tail(&rcache->lru.list, &region->lru_list);
     ++rcache->lru.count;
+    ucs_spin_lock(&rcache->lock);
```

Comment:
but this function comment says `/* LRU spinlock must be held */`
problem is where we add/remove other non-lru flags
that's why I made it a standalone field..

Reply:
Understood. I reverted the flags back to be a separate variable. As suggested in https://github.com/openucx/ucx/pull/6434#discussion_r587288379, I made the flags an enum to allow future extension.